whois-nrtm4/src/main/java/net/ripe/db/nrtm4/NrtmFileStore.java

```diff
@@ -0,0 +1,52 @@
+package net.ripe.db.nrtm4;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+
+
+@Service
+public class NrtmFileStore {
+
+    private final String path;
+
+    NrtmFileStore(
+        @Value("${nrtm.file.path:/tmp}") final String path
```

Comment:
this is the path to the dir where the files are stored

Reply:
yes, but I don't understand why we need to store data in a file. Maybe we talk about it before going on holidays but I don't remember now :(
Could you explain me why we need to use a file if we already have the data in the database?

Reply:
the idea is to reduce the workload on the server in case we get a lot of requests. if we can serve a file from the file system it's quicker than going to the database for it, so we use the file system as a kind of cache. when it's working i think it's a good idea to do some load tests to see how well this works. it won't help us in the case of a DoS attack, though. maybe we need a different countermeasure to help with that (e.g. put all valid sessionID/filenames in memory? just an idea).

Reply:
Yes, maybe we need to think a little bit more about this. Maybe use @Cacheable and store data in memory and build the output using the cached data? there is a way to create a programatic cache (spring boot at least) I have done some of them, very useful for Collections maybe we should keep an eye on this.